### DISCUSSION TITLE
Do I need to use Smee?

### DISCUSSION THREAD

**Discussion by Html-superfan** [community/community#24423] [Apps, API and Webhooks]
<p>What if I already have ngrok or localtunnel installed?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, you don’t have to use Smee. Both ngrok and localtunnel will work just fine if you prefer them. However, Smee has several advantages for first-time GitHub App developers that make the steps in this Quickstart easier to follow. For example, Smee runs entirely on its own server. Smee also gives you a unique domain that will never stop working for your app, so you can stop and start the Smee client without having to update your app settings with a new domain.</p>

**Comment by levimatheri** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> Is it ever possible that two separate apps would have the same smee url? I imagine it’s highly unlikely since there is 36^15 possibilities/</p>

**Comment by lee-dohm** [comment]
<p>If you use the randomly generated ID, then you are correct. But there’s nothing preventing you from picking your own ID <a href="https://smee.io/my-really-cool-app" rel="nofollow noopener">for example</a>. Doing a quick search on the repository though, I found <a href="https://github.com/probot/smee.io/issues/17" rel="nofollow noopener">this issue</a> which may make the situation clearer.</p>
<p>I hope that helps!</p>

**Comment by amhaqawi** [comment]
<p>Thank you my friend</p>
